component/azstorage/datalake.go

```diff
@@ -481,90 +444,103 @@ func (dl *Datalake) List(prefix string, marker *string, count int32) ([]*interna
 	}
 
 	// Get a result segment starting with the path indicated by the current Marker.
-	listPath, err := dl.Filesystem.ListPaths(context.Background(),
-		azbfs.ListPathsFilesystemOptions{
-			Path:              &prefixPath,
-			Recursive:         false,
-			MaxResults:        &count,
-			ContinuationToken: marker,
-		})
+	listPath := dl.Filesystem.NewListPathsPager(false, &filesystem.ListPathsOptions{
+		Marker:     marker,
+		MaxResults: &count,
+		Prefix:     &prefixPath,
+	})
+
+	//TODO:: track2: review below
+	// if err != nil {
+	// 	log.Err("Datalake::List : Failed to validate account with given auth %s", err.Error())
+	// 	m := ""
+	// 	e := storeDatalakeErrToErr(err)
+	// 	if e == ErrFileNotFound { // TODO: should this be checked for list calls
+	// 		return pathList, &m, syscall.ENOENT
+	// 	} else if e == InvalidPermission {
+	// 		return pathList, &m, syscall.EACCES
+	// 	} else {
+	// 		return pathList, &m, err
+	// 	}
+	// }
 
-	if err != nil {
-		log.Err("Datalake::List : Failed to validate account with given auth %s", err.Error())
-		m := ""
-		e := storeDatalakeErrToErr(err)
-		if e == ErrFileNotFound { // TODO: should this be checked for list calls
-			return pathList, &m, syscall.ENOENT
-		} else if e == InvalidPermission {
-			return pathList, &m, syscall.EACCES
-		} else {
-			return pathList, &m, err
+	var m *string
+	// Process the paths returned in this result segment (if the segment is empty, the loop body won't execute)
+	for listPath.More() {
```

Comment:
List() method should not iterate over the pages. It should make call to the NextPage() once.

Reply:
Refer the List() method in blockblob.go